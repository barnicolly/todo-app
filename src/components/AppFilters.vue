<template>
  <aside class="todo-list__filters">
    <section class="toggle-group">
      <button
        v-for="filter in filters"
        :key="filter"
        class="toggle-group__button"
        :class="{ 'toggle-group__button--primary': activeFilter === filter }"
        @click="setFilter(filter)">
        {{ filter }}
      </button>
    </section>
  </aside>
</template>

<script setup lang="ts">
import { Filter } from '@/types/filter';
import { ref } from 'vue';
import { useTodoStore } from '@/store/todo';
import { storeToRefs } from 'pinia';

const store = useTodoStore();
const { filter: activeFilter } = storeToRefs(store);

const filters = ref([Filter.All, Filter.Active, Filter.Done]);

function setFilter(filter: Filter): void {
  store.setFilter(filter);
}
</script>
