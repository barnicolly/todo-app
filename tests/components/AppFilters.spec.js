import { shallowMount } from '@vue/test-utils';
import AppFilters from '@/components/AppFilters.vue';
import { createTestingPinia } from '@pinia/testing';

describe('AppFilters.vue', () => {
  it('should correct initialize', () => {
    const wrapper = shallowMount(AppFilters, {
      props: {
        activeFilter: 'All',
      },
      global: {
        plugins: [createTestingPinia({ stubActions: false })],
      },
    });

    const button = wrapper.find('.toggle-group__button:nth-child(1)');
    expect(wrapper.vm.activeFilter).toBe('All');
    expect(wrapper.vm.filters).toStrictEqual(['All', 'Active', 'Done']);
    expect(button.element.classList).toContain('toggle-group__button--primary');
  });

  it('click active filter', async () => {
    const wrapper = shallowMount(AppFilters, {
      props: {
        activeFilter: 'All',
      },
      global: {
        plugins: [createTestingPinia({ stubActions: false })],
      },
    });

    const activeFilterButton = wrapper.find(
      '.toggle-group__button:nth-child(2)',
    );
    await activeFilterButton.trigger('click');
    expect(activeFilterButton.element.classList).toContain(
      'toggle-group__button--primary',
    );
    expect(wrapper.vm.activeFilter).toBe('Active');
  });
});

describe('AppFilters.vue Snapshot', () => {
  it('has same HTML structure', () => {
    const wrapper = shallowMount(AppFilters, {
      props: {
        activeFilter: 'All',
      },
      global: {
        plugins: [createTestingPinia({ stubActions: false })],
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
