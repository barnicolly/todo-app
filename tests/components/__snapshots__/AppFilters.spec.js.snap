// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppFilters.vue Snapshot has same HTML structure 1`] = `
<aside class="todo-list__filters" activefilter="All">
  <section class="toggle-group"><button class="toggle-group__button toggle-group__button--primary">All</button><button class="toggle-group__button">Active</button><button class="toggle-group__button">Done</button></section>
</aside>
`;
